File: 4_median_of_two_sorted_arrays.c
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int first = 0, second = 0;
    int merged_index = 0, merged_size = nums1Size + nums2Size;
    int merged[merged_size];

    while ((first < nums1Size) && (second < nums2Size)) {
        if (nums1[first] < nums2[second]) {
            merged[merged_index] = nums1[first];
            first++;
        }
        else {
            merged[merged_index] = nums2[second];
            second++;
        }
        merged_index++;
    }
    
    for (int i = first; i < nums1Size; i++) {
        merged[merged_index] = nums1[i];
        merged_index++;
    }
    for (int i = second; i < nums2Size; i++) {
        merged[merged_index] = nums2[i];
        merged_index++;
    }

    float median;
    if (merged_size % 2 != 0) {
        median = merged[merged_size / 2] / 1.0;
    }
    else {
        int a = merged[merged_size / 2];
        int b = merged[(merged_size / 2) - 1];
        median = (a + b) / 2.0;
    }
    return median;
}

File: 4_median_of_two_sorted_arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        if len(nums1) < len(nums2):
            small = nums1
            big = nums2
        else:
            small = nums2
            big = nums1

        first, second = 0, 0
        merged = [] 
        while (first < len(small)) and (second < len(big)):
            if small[first] < big[second]:
                merged.append(small[first])
                first += 1
            else:
                merged.append(big[second])
                second += 1
        merged = merged + small[first:]
        merged = merged + big[second:]

        median = merged[len(merged)//2] if len(merged) % 2 != 0 else (merged[len(merged)//2] + merged[(len(merged)//2) - 1]) / 2
        return float(median)